### PR TITLE
docs: Improve development guide.

### PR DESCRIPTION
**Pull request by vsvipul**
Minifies the current development guide by
a significant amount. Also, removes system requirements
as no credible source for them. Removes dependency
specific instructions and adds links to instructions
to install them.

Preview [here](https://github.com/zulip/zulip-electron/blob/0f96d06fbcec36fefaf979cc1f91805e1e4e2119/development.md).

**You have tested this PR on:**
  - [x] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

File: development.md [hunk 1]
```diff
@@ -1,152 +1,109 @@
...
+* [Git](http://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+  * Use our [Git Guide](https://zulip.readthedocs.io/en/latest/git/setup.html) to get started with Git and GitHub.
+* [Node.js](https://nodejs.org) >= v6.9.0
+  * [NPM](https://www.npmjs.com/get-npm) and
+    [node-gyp](https://github.com/nodejs/node-gyp#installation),
+    if they don't come bundled with your Node.js installation
+* [Python](https://www.python.org/downloads/release/python-2713/)
+  (v2.7.x recommended)
+* A C++ compiler compatible with C++11
+* Linux users also need [Snapcraft](https://snapcraft.io/)
```
Comment (abhigyank) [inline development.md:19]: Snapcraft is not necessary for development. Can remove this.

File: development.md [hunk 2]
```diff
@@ -1,152 +1,109 @@
...
+* Development headers for the libXext, libXtst, and libxkbfile libraries
```
Comment (abhigyank) [inline development.md:20]: Can include the command for it somewhere, at least in Linux `$ sudo apt install libxext-dev libxtst-dev libxkbfile-dev libgconf-2-4`.

File: development.md [hunk 3]
```diff
@@ -1,152 +1,109 @@
...
+install nodejs. To ensure Node.js has been installed, run  ```node -v``` in terminal to know your node version.
 
-## MacOS
+[node-mac]: https://nodejs.org/en/download/package-manager/#macos
 
+If [NPM](https://www.npmjs.com/get-npm) and [node-gyp](https://github.com/nodejs/node-gyp#installation) don't come bundled with your Node.js installation, you will need to install them manually. 
 
-  **Node JS**
-  Go to the [Node.js Downloads page](https://nodejs.org/en/download/). Download Node.js for MacOS (`v6.9.0` or above recommended). Run the downloaded Node.js `.pkg` Installer. You're finished! To ensure Node.js has been installed, run `node -v` in your terminal - you should get something like `v6.9.0` or above 
+### Windows
```
Comment (abhigyank) [inline development.md:44]: On Windows, I think we need to install some global dependencies as well. Probably - `npm install --global --production windows-build-tools` as admin.

File: development.md (hunk 2 above)
Comment (abhigyank) [reply development.md:20]: Do we also need to add `$ sudo apt install build-essential`?

**Comment by abhigyank** [comment]
@kanishk98 is there anything else needed for in the Windows Dev?

**Comment by akashnimare** [comment]
@vsvipul let's finish this.

**Comment by vsvipul** [comment]
@akashnimare closing this, will open a new PR.

**Pull request closed without merging**